### DISCUSSION TITLE
Migrate yarn.lock dependencies to bun.lockb file

### DISCUSSION THREAD

**Discussion by haseeb-ul-hassan-123** [community/community#136325] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I want to read yarn.lock file and generate bun.lockb from that with the same dependencies to be used in bun install, without this bun is creating the deps based on package.json and is big projects (specially typescript projects) it brings lost of type conflicts.

**Comment by hi-ammad** [comment]
Bun (since version [1.0.5](https://github.com/oven-sh/bun/releases/tag/bun-v1.0.5)) supports migrating package-lock.json to bun.lockb as seen in [this pull request](https://github.com/oven-sh/bun/pull/6352). However, the support for migrating yarn.lock is not implemented yet. You can track the progress on [this issue](https://github.com/oven-sh/bun/issues/6409).

As an intermediate step, we can use [synp](https://github.com/imsnif/synp), a tool that converts yarn.lock to package-lock.json.
Here's the detailed steps:
```
# convert yarn.lock to package-lock.json using synp
npx synp --source-file yarn.lock

# upgrade package-lock.json to version 3 using npm
npm i --lockfile-version 3 --frozen-lockfile

# convert package-lock.json to bun.lockb using bun
bun pm migrate
```
